mixer/pkg/cache/lruCache.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cache
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// We keep a slice of entries representing all items currently in the cache.
+// The entries in this slice are organized in a doubly-linked circular list. The list
+// is created via next/prev fields that contain indices into the slice. Using a slice
+// with indices in this way is considerably more efficient, in both time and space, than
```

Comment:
should you add "...considerably more efficient **than list.List**..." to be clear about what the comparison is to?

Reply:
I think the next line of text addresses that, no?

Reply:
i guess. yeah, i'm starting to wonder when anyone would ever user `list.List`.

Reply:
Implementing an efficient linked list abstraction in a managed language is difficult. The way the golang library chose to do it is definitely not the efficient choice :-(